docs/modules/ROOT/pages/dev_guide.adoc

```diff
@@ -1,22 +1,50 @@
 = Development Guide
 
-== Automation
+== Release
 
-=== Release Process
+=== Preparation
 
-Release automation is based on a tagged commit (i.e. named `release: v0.0.1` with associated annotated tag `v0.0.1`)
-Such a commit should have the following content:
+Before cutting the release we should prepare notes about its content. Assuming you are releasing `v0.0.2`, these are the steps you should perform:
 
-. The version in `docs/antora.yml` changed to a matching `tag`
-. Release notes named after tag `tag.adoc` in `docs/modules/ROOT/pages/release_notes/`
-. New release included in `docs/modules/ROOT/pages/release_notes.adoc`
+. Draft highlights of the release (most notable features etc.). Use the template `docs/modules/ROOT/pages/release_notes/release_notes_template.adoc`
+. Name the release notes file the same as new version and put it in `docs/modules/ROOT/pages/release_notes/v0.0.2.adoc` folder
+. Create new branch called `release_v0.0.2`
+. Open Pull Request so we can review the announcement and other related changes
 
-NOTE: Above steps are automated by `scripts/release.sh`
+Once the PR is approved you can perform additional steps by invoking `scripts/release.sh` as follows:
 
-Such tagged commit triggers a job in CircleCI which will:
+[source,bash]
+----
+./scripts/release.sh -v v0.0.2 --dry-run
+----
 
-. Build and push tagged docker image to `quay.io` registry
-. Push cross-compiled binaries and release notes to GitHub
+This will validate if above steps were correctly performed and create two commits:
+
+. `release: v0.0.2`
+  * in this commit autogenerated changelog will be appended to release notes
+  * the version in `docs/antora.yml` will be changed to a matching version
+  * release notes will be prepended to `docs/modules/ROOT/pages/release_notes.adoc`
+. `release: next iteration`
+  * reverts version-related changes from previous commit
+
+IMPORTANT: Not using `--dry-run` flag will result in pushing release commits.
```

Comment:
Just a question; Since running this is part of the manual process, and it comes with a warning.. Couldn't we just switch it around: If you run it without a --do-release it does a dry run. That way it's a more conscious action to make the release (it will be scripted anyway so 'we' won't use the do-release flag.)

Reply:
Agreed, lemme add that part. Generally speaking, I want that to be fully automated soon, but until then... :)

Reply:
e4874be